plugin/environment.go

```diff
@@ -113,13 +115,22 @@ func (env *Environment) Statuses() (model.PluginStatuses, error) {
 			pluginState = plugin.(activePlugin).State
 		}
 
+		pluginErr := ""
+		if pluginState == model.PluginStateNotRunning {
+			if err := env.CheckPluginHealthStatus(plugin.Manifest.Id); err != nil {
+				pluginErr = err.Error()
+				pluginState = model.PluginStateError
+			}
```

Comment:
I'm thinking this may need to be i18n'd since these errors would be returning to the UI. The kind of error that would be here is something like "Plugin process not found, or not responding", or something returned from the HealthCheck hook.

This PR has been mainly focused on the auto-restart capabilities of the plugins, and seems to have shifted some focus away from the UX from the admin console of managing the failed plugins. Any thoughts on making the `api/v4/plugins/<plugin_id>/restart` api route, and the UX around including error info in the `api/v4/plugins/<plugin_id>/statuses` route into a separate PR to keep the scope of this PR more isolated? However, that PR would depend on this PR's functionality.

Reply:
There's some more plugin framework work to be done here as well, like docs and testing of the HealthCheck hook, and possibly adding the `RestartPlugin` method to the PluginAPI.

Reply:
Ya. If you want to do the UI as a separate PR so this can get merged. That would work.

Reply:
@crspeller Would you suggest I remove the restart api route from this PR, or keep it how it is and continue development on it when this is merged?